### PR TITLE
fix: fix up ep resourceVersion comparison and clean up

### PR DESCRIPTION
**Pull request by cmssczy**
### Type of change:

- [x] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches

### Pre-submission checklist:

fix https://github.com/apache/apisix-ingress-controller/issues/806#issuecomment-1059945155
* [ ] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix-ingress-controller#community) first**

### CODE REVIEW

**Review by tao12345666333 — APPROVED** [review]

File: pkg/ingress/status.go [hunk 1]
```diff
@@ -28,7 +28,6 @@ import (
 	networkingv1beta1 "k8s.io/api/networking/v1beta1"
 	"k8s.io/apimachinery/pkg/api/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Comment (cmssczy) [inline pkg/ingress/status.go:31]: duplicated

**Review by lingsamuel — APPROVED** [review]

**Comment by cmssczy** [comment]
updated: apisixTls/ endpointSlice/ secret aslo have this problem. 
@tao12345666333 @lingsamuel

**Comment by tao12345666333** [comment]
good catch.

**Review by gxthrj — APPROVED** [review]

**Review by lingsamuel — APPROVED** [review]

**Comment by cmssczy** [comment]
maybe need to re-run e2e-test😂 @tao12345666333

**Comment by tao12345666333** [comment]
re-run CI.

**Comment by tao12345666333** [comment]
It's green! Thanks!

**Pull request merged**